7/README.md

```diff
@@ -0,0 +1,95 @@
+---
+domain: rfc.tango-controls.org 
+shortname: 7/PIPE 
+name: The pipe 
+status: raw 
+editor:  
+contributors: Olga Merkulova `<at ingvord.ru - olga>`, Igor Khokhriakov `<at ingvord.ru - mail>`
+---
+
+## Preamble
+
+Copyright (c) 2019 Tango Controls
+
+This Specification is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 3 of the License, or (at your option) any later version. This Specification is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with this program; if not, see <http://www.gnu.org/licenses>.
+
+This Specification is a [free and open standard](http://www.digistan.org/open-standard:definition) and is governed by the Digital Standards
+Organization's [Consensus-Oriented Specification System](http://www.digistan.org/spec:1/COSS).
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+
+# Tango Pipe specificatiom
+
+This specification describes the Tango Pipe feature of a Device. 
+
+## Goals
+
+Tango Pipe is an Attribute [RFC-4] with custom data structure.
+
+## Use Case
+
+1. To pass big pieces of data structure at a time
+2. To define complex data structures to act as a single Attribute
+
+## Specification
+
+Device MAY define zero, one or more Pipes.
+
+Pipe MUST have static metadata:
+
+ * name
+ * description
+ * label
+ * level
+ * writeType
+ * writable
+
+```abnf
+name = TangoName
+
+description = DevString [link to RFC-9 Data Types]
+
+label = DevString [link to RFC-9 Data Types]
+
+level = "OPERATOR"/"USER"
+
+writeType = "PIPE_READ"/"PIPE_WRITE"
+
+writable = DevBoolean
+```
+
+Pipe name MUST follow Tango naming conventions.
+
+Pipe with writeType="PIPE_READ" MUST be read only.
+
+Pipe data structure MUST be:
+
+ * Key(s) -> Value(s)
+
+```abnf
+Key = DevString [link to RFC-9 Data Types]
+
+Value = An Arary of Any Tango compliant type or Pipe [link to RFC-9 Data Types]
```

Comment:
@bourtemb Thank you for the comments. 

I've made changes. And I have added an image with "Key-Value" based on the Figure 4.1 https://tango-controls.readthedocs.io/en/latest/development/client-api/cpp-client-programmers-guide.html#reading-writing-device-pipe

The reason is that I don't know whether it is a good idea to use such special words as  "DataElement" or "PipeBlob" in RFC... As I understand, in fact it is something like Map so why not to use common words.

Reply:
Thanks @HelgaMond for the update.
I think there is an important information missing in the new image you uploaded.
On the right, in the figure from the doc, the data element value can be data or a DevicePipeBlob.
In the new image, data is absent  (This is actually described in the abnf definition of Value you proposed). So this makes the structure appearing to be infinitely recursive. I find it a bit confusing.
The key-value analogy is not stupid I think but I would not say that a pipe is completely like a map. It is more a special structure. I personally don't like the term "Description" at the higher level. I find it confusing too.
In my opinon, the terms DevicePipeBlob, DataElement are already mentioned in the current documentation and in the Tango IDL (https://github.com/tango-controls/tango-idl/blob/tango-9-lts/tango.idl#L547-L569) so I would simply use these terms.

Reply:
@HelgaMond, I noticed @piogor described the pipes in RFC-9 (Datatypes) too (https://github.com/tango-controls/rfc/pull/61/files), so if you change the names, it would be good to coordinate with @piogor to get some consistent names across the different RFCs.

Reply:
@bourtemb Could you have a look at the RFC-7, please? I have made minor changes and as agreed have returned to the "DevicePipeBlob" definition. Thank you!